tests/fuzz/s2n_openssl_diff_pem_parsing_test.c

```diff
@@ -93,17 +97,41 @@ int LLVMFuzzerTestOneInput(const uint8_t *buf, size_t len)
     memset(&config->cert_and_key_pairs->ocsp_status, 0, sizeof(config->cert_and_key_pairs->ocsp_status));
     memset(&config->cert_and_key_pairs->sct_list, 0, sizeof(config->cert_and_key_pairs->sct_list));
 
-    int parsable_by_s2n = !s2n_config_add_cert_chain(config, (const char*) in.blob.data);
+    /* Use s2n_config_add_cert_chain_from_stuffer() so that \0 characters don't truncate strings. */
+     s2n_config_add_cert_chain_from_stuffer(config, in);
+
+    struct s2n_cert *next = config->cert_and_key_pairs->cert_chain.head;
+    int chain_len = 0;
+    while(next != NULL) {
+        chain_len++;
+        next = next->next;
+    }
 
-    GUARD(s2n_stuffer_free(&in));
     GUARD(s2n_config_free(config));
 
-    if (parsable_by_openssl && !parsable_by_s2n) {
+    return chain_len;
+}
+
+int LLVMFuzzerTestOneInput(const uint8_t *buf, size_t len)
+{
+    struct s2n_stuffer in;
+    GUARD(s2n_stuffer_alloc(&in, len + 1));
+    GUARD(s2n_stuffer_write_bytes(&in, buf, len));
+    in.blob.data[len] = 0;
+
+    uint8_t openssl_chain_len = openssl_parse_cert_chain(&in);
+    GUARD(s2n_stuffer_reread(&in));
+    uint8_t s2n_chain_len = s2n_parse_cert_chain(&in);
+    GUARD(s2n_stuffer_free(&in));
+
+
+    if (openssl_chain_len > s2n_chain_len) {
```

Comment:
I think this is fine to merge. One idea I had: Should we also require that the parsed certificate would result in a successful handshake? I don't think sucessfully *parsed* but *unuable* inputs are particularly interesting.